alerting_alert_rule.go

```diff
@@ -0,0 +1,95 @@
+package gapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// AlertRule represents a Grafana Alerting Alert Rule.
+type AlertRule struct {
+	Annotations  map[string]string `json:"annotations,omitempty"`
+	Conditions   string            `json:"conditions"`
+	Data         []*AlertQuery     `json:"data"`
+	ExecErrState ExecErrState      `json:"execErrState"`
+	FolderUID    string            `json:"folderUid"`
+	ID           int64             `json:"id,omitempty"`
+	Labels       map[string]string `json:"labels,omitempty"`
+	NoDataState  NoDataState       `json:"noDataState"`
+	OrgID        int64             `json:"orgId"`
+	RuleGroup    string            `json:"ruleGroup"`
+	Title        string            `json:"title"`
+	UID          string            `json:"uid,omitempty"`
+	Updated      string            `json:"updated,omitempty"`
+	ForDuration  time.Duration     `json:"for"`
+	Provenance   string            `json:"provenance"`
+}
+
+type AlertQuery struct {
+	DatasourceUID     string
+	Model             interface{}
+	QueryType         string
+	RefID             string
+	RelativeTimeRange RelativeTimeRange
```

Comment:
```suggestion
	DatasourceUID     string `json:"datasourceUid,omitempty"`
	Model             interface{} `json:"model"`
	QueryType         string `json:"queryType,omitempty"`
	RefID             string `json:"refId,omitempty"`
	RelativeTimeRange RelativeTimeRange `json:"relativeTimeRange"`
```
Just for consistency.